### PR TITLE
Feature/workallocation/location data service

### PR DESCRIPTION
**Pull request by connorpgpmcelroy**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-2872

### Change description ###
Creating location-data-service as per (caseworker/task)-data-service

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by connorpgpmcelroy** [comment]
Merged Phil's changes to the node layer in the api files

**Comment by connorpgpmcelroy** [comment]
https://tools.hmcts.net/jira/browse/EUI-2873

File: api/test/pact/constants/behaviors.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export const SUCCESS = {};
```
Comment (paul-graham) [inline api/test/pact/constants/behaviors.ts:1]: I think this should be `behaviors.spec.ts` so it's not compiled into the code. It's purely for testing.

Comment (paul-graham) [reply api/test/pact/constants/behaviors.ts:1]: Actually, "behaviours" seeing as we're not American. :)

Comment (connorpgpmcelroy) [reply api/test/pact/constants/behaviors.ts:1]: @ficklephil are you happy for me to change the filename?

Comment (ficklephil) [reply api/test/pact/constants/behaviors.ts:1]: @connorpgpmcelroy Yep go for it :)

File: api/test/pact/pact-tests/work-allocation-get-locations.spec.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  // Write Pact when all tests done
+  after(() => provider.finalize());
+
+  /**
+   * Note that the location body is yet to be defined, by us or WA Allocation team.
+   */
+  describe('should return locations.', () => {
+
+    const LOCATION_BODY = ['location', 'location'];
```
Comment (paul-graham) [inline api/test/pact/pact-tests/work-allocation-get-locations.spec.ts:36]: Isn't the location body the one defined in `api/workAllocation/interfaces/task.ts`, line 30?

Comment (ficklephil) [reply api/test/pact/pact-tests/work-allocation-get-locations.spec.ts:36]: You are right, I missed that, will put in the changes.

Comment (ficklephil) [reply api/test/pact/pact-tests/work-allocation-get-locations.spec.ts:36]: You are right, I missed that, will put in the changes.

Comment (ficklephil) [reply api/test/pact/pact-tests/work-allocation-get-locations.spec.ts:36]: @connorpgpmcelroy has kindly offered to put in the changes, thank you!

Comment (connorpgpmcelroy) [reply api/test/pact/pact-tests/work-allocation-get-locations.spec.ts:43]: Should be all good now

File: api/test/pact/pact-tests/work-allocation-get-locations.spec.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+import { EnhancedRequest } from '../../../lib/models';
+import { handleLocationGet } from '../../../workAllocation/locationService';
+
+describe('Work Allocation Location API', () => {
+
+  let mockServerPort: number
+  let provider: Pact
+  const location = [
+    {
+        "id": "string",
```
Comment (paul-graham) [inline api/test/pact/pact-tests/work-allocation-get-locations.spec.ts:15]: Single quotes. Sorry.

**Review by udaydenduluri33 — APPROVED** [review]
LGTMc&p

File: api/test/pact/pact-tests/work-allocation-get-locations.spec.ts (hunk 3 above)
Comment (connorpgpmcelroy) [reply api/test/pact/pact-tests/work-allocation-get-locations.spec.ts:15]: that's cool, I will sort

**Review by paul-graham — APPROVED** [review]
Looks good.

**Pull request merged**